### PR TITLE
Release 2.3.14

### PR DESCRIPTION
**Pull request by HanlinMiao**
- [x] Towncrier
- [x] Version Bump

PRs currently in flight:
- [x] https://github.com/nautobot/nautobot/pull/6675
- [x] https://github.com/nautobot/nautobot/pull/6673
- [x] https://github.com/nautobot/nautobot/pull/6682
- [x] https://github.com/nautobot/nautobot/pull/6681
- [x] https://github.com/nautobot/nautobot/pull/6680
- ~https://github.com/nautobot/nautobot/pull/6672~ (will be included in 2.3.15)
- ~https://github.com/nautobot/nautobot/pull/6579 (Community PR)~ needs further refinement or an alternate approach, see https://github.com/nautobot/nautobot/pull/6579#issuecomment-2521397548

### CODE REVIEW

File: nautobot/docs/release-notes/version-2.3.md [hunk 1]
```diff
@@ -124,6 +124,33 @@ Various button groups in the "object list" and "object detail" views have been c
 As Django 3.2 has reached end-of-life, Nautobot 2.3 requires Django 4.2, the next long-term-support (LTS) version of Django. There are a number of changes in Django itself as a result of this upgrade; Nautobot App maintainers are urged to review the Django release-notes ([4.0](https://docs.djangoproject.com/en/4.2/releases/4.0/), [4.1](https://docs.djangoproject.com/en/4.2/releases/4.1/), [4.2](https://docs.djangoproject.com/en/4.2/releases/4.2/)), especially the relevant "Backwards incompatible changes" sections, to proactively identify any impact to their Apps.
 
 <!-- towncrier release notes start -->
+## v2.3.14 (2024-12-18)
+
+### Added
+
+- [#6548](https://github.com/nautobot/nautobot/issues/6548) - Added logic to set the parent in the clean() method of the Prefix model, ensuring correct assignment during validation.
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.3.md:131]:
```suggestion
- [#6548](https://github.com/nautobot/nautobot/issues/6548) - Added logic to set the `parent` in the `clean()` method of the Prefix model, ensuring correct assignment during validation.
```

File: nautobot/docs/release-notes/version-2.3.md [hunk 2]
```diff
@@ -124,6 +124,33 @@ Various button groups in the "object list" and "object detail" views have been c
...
+
+### Changed
+
+- [#6518](https://github.com/nautobot/nautobot/issues/6518) - Added VRF's column to Prefixes and Child Prefixes table.
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.3.md:135]: Should this be .added?

```suggestion
- [#6518](https://github.com/nautobot/nautobot/issues/6518) - Added VRFs column to Prefixes and Child Prefixes tables.
```

File: nautobot/docs/release-notes/version-2.3.md [hunk 3]
```diff
@@ -124,6 +124,33 @@ Various button groups in the "object list" and "object detail" views have been c
...
+- [#6548](https://github.com/nautobot/nautobot/issues/6548) - Changed the save method of the `Prefix` model to reparent subnets and IPs only when the `network`, `namespace`, or `prefix_length` fields are updated.
+
+### Fixed
+
+- [#6204](https://github.com/nautobot/nautobot/issues/6204) - Fixed out-of-memory errors when `LogsCleanup` system job resulted in cascade deletion of many related objects (such as `JobLogEntry` or `nautobot_ssot.SyncLogEntry` records).
+- [#6496](https://github.com/nautobot/nautobot/issues/6496) - Fixed `/ipam/prefixes/<UUID>/available-ips` to calculate IP's with IP's from Child Prefixes.
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.3.md:141]:
```suggestion
- [#6496](https://github.com/nautobot/nautobot/issues/6496) - Fixed `/ipam/prefixes/<UUID>/available-ips/` to correctly consider IPs under child Prefixes.
```

File: nautobot/docs/release-notes/version-2.3.md [hunk 4]
```diff
@@ -124,6 +124,33 @@ Various button groups in the "object list" and "object detail" views have been c
...
+- [#6496](https://github.com/nautobot/nautobot/issues/6496) - Fixed `Prefix.get_first_available_ip` method to not return IP taken by Child Prefixes.
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.3.md:142]:
```suggestion
- [#6496](https://github.com/nautobot/nautobot/issues/6496) - Fixed `Prefix.get_first_available_ip()` method to not return IP taken by child Prefixes.
```

**Review by glennmatthews — APPROVED** [review]

**Review by lampwins — APPROVED** [review]

**Pull request merged**